api/src/services/graphql/index.ts

```diff
@@ -1405,6 +1413,16 @@ export class GraphQLService {
 
 		const result = await this.read(collection, query);
 
+		if (args['version']) {
+			const versionsService = new VersionsService({ accountability: this.accountability, schema: this.schema });
+
+			const saves = await versionsService.getVersionSaves(args['version'], collection, args['id']);
+
+			if (saves) {
+				return assign(result?.[0] || {}, ...saves);
+			}
```

Comment:
Differentiation required for singleton versions
```suggestion
			if (saves) {
				const item = this.schema.collections[collection]!.singleton ? result : result?.[0] || {};

				return assign(item, ...saves);
			}
```


Also should we return `null` if `result?.[0]` is undefined, similar to https://github.com/directus/directus/pull/19910/files#diff-a3189ece0cf91df6772d5b5390785b080eab1607e2c40518425f968b3ffcfcadR1427 ?